tests/test_templates.py

```diff
@@ -0,0 +1,23 @@
+import glob
+import pytest
+import json
+import os
+
+revpi_hat_templates = sorted(glob.glob("./revpi-hat-PR*.json"))
+
+
+@pytest.mark.parametrize("template", revpi_hat_templates)
+def test_filename(template: str):
+    with open(template, "r") as f:
+        template_data = json.loads(f.read())
+
+        product_id = template_data["pid"] + 100000
+        product_revision = template_data["prev"]
+
+        filename = os.path.basename(template)
+        expected_filename = f"revpi-hat-PR{product_id}R{product_revision:02}.json"
+
+        assert filename == expected_filename, (
```

Comment:
No, it won't:

```
➜  revpi-hat-data git:(devel/quality-checks) ✗ pytest --tb=no -v --junit-xml report.xml
============================================================================================================== test session starts ==============================================================================================================
platform linux -- Python 3.9.2, pytest-7.2.0, pluggy-1.0.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/nbuchwitz/dev/revpi/revpi-hat-data
plugins: anyio-3.6.2
collected 16 items                                                                                                                                                                                                                              

tests/test_templates.py::test_filename[./revpi-hat-PR100299R01.json] FAILED                                                                                                                                                               [  6%]
tests/test_templates.py::test_filename[./revpi-hat-PR100300R01.json] PASSED                                                                                                                                                               [ 12%]
tests/test_templates.py::test_filename[./revpi-hat-PR100301R01.json] PASSED                                                                                                                                                               [ 18%]
tests/test_templates.py::test_filename[./revpi-hat-PR100328R03.json] PASSED                                                                                                                                                               [ 25%]
tests/test_templates.py::test_filename[./revpi-hat-PR100359R01.json] PASSED                                                                                                                                                               [ 31%]
tests/test_templates.py::test_filename[./revpi-hat-PR100360R01.json] PASSED                                                                                                                                                               [ 37%]
tests/test_templates.py::test_filename[./revpi-hat-PR100361R01.json] PASSED                                                                                                                                                               [ 43%]
tests/test_templates.py::test_filename[./revpi-hat-PR100365R00.json] PASSED                                                                                                                                                               [ 50%]
tests/test_templates.py::test_filename[./revpi-hat-PR100366R00.json] PASSED                                                                                                                                                               [ 56%]
tests/test_templates.py::test_filename[./revpi-hat-PR100367R00.json] PASSED                                                                                                                                                               [ 62%]
tests/test_templates.py::test_filename[./revpi-hat-PR100375R00.json] PASSED                                                                                                                                                               [ 68%]
tests/test_templates.py::test_filename[./revpi-hat-PR100376R00.json] PASSED                                                                                                                                                               [ 75%]
tests/test_templates.py::test_filename[./revpi-hat-PR100377R00.json] PASSED                                                                                                                                                               [ 81%]
tests/test_templates.py::test_filename[./revpi-hat-PR100378R00.json] PASSED                                                                                                                                                               [ 87%]
tests/test_templates.py::test_filename[./revpi-hat-PR100379R00.json] PASSED                                                                                                                                                               [ 93%]
tests/test_templates.py::test_filename[./revpi-hat-PR100380R00.json] PASSED                                                                                                                                                               [100%]

------------------------------------------------------------------------------------ generated xml file: /home/nbuchwitz/dev/revpi/revpi-hat-data/report.xml ------------------------------------------------------------------------------------
============================================================================================================ short test summary info ============================================================================================================
FAILED tests/test_templates.py::test_filename[./revpi-hat-PR100299R01.json] - AssertionError: Template filename (revpi-hat-PR100299R01.json) does not match expected filename (revpi-hat-PR102991R01.json) based on the product details
========================================================================================================= 1 failed, 15 passed in 0.05s ==========================================================================================================
```

This is because of `("template", revpi_hat_templates)` treats each file a separate test.